pkg/endpoints/file/download.go

```diff
@@ -24,20 +24,23 @@ func DownloadEndpoint(store db.FileStore) func(w http.ResponseWriter, r *http.Re
 			log.Println("key parameter required for file download request")
 		}
 
-		fmt.Println(keys)
+		id, err := strconv.ParseInt(keys[0], 10, 64)
+		if endpoints.HttpNotOk(400, w, "invalid document id", err) {
+			return
+		}
 
 		w.Header().Set("Content-Type", "application/json")
 
-		contents, err := store.GetFile(keys[0])
+		contents, fileSpec, err := store.GetFile(int(id))
 
 		if err != nil {
 			log.Println(err)
 			w.WriteHeader(400)
 		} else {
 			filetype := parser.Text
-			if strings.HasSuffix(keys[0], ".docx") {
+			if fileSpec.Ext == "docx" {
 				filetype = parser.DocX
-			} else if strings.HasSuffix(keys[0], ".xlsx") {
+			} else if fileSpec.Ext == "xlsx" {
 				filetype = parser.Xlsx
 			}
```

Comment:
if we start supporting more file types (besides csv), doing this check in different places is going to get old.
Of course, dunno what file type besides .csv we might try to support... plain .Xls *mabye*?

Reply:
Moved to method for now, I have a feeling we might need to rethink some stuff about file uploads/downloads, but at least in a function its separated out from other stuff.